HTML autocorrect attribute (#35692)
mdn/content: files/en-us/web/html/global_attributes/autocorrect/index.md

---
title: autocorrect
slug: Web/HTML/Global_attributes/autocorrect
page-type: html-attribute
browser-compat: html.global_attributes.autocorrect
---

{{HTMLSidebar("Global_attributes")}}

The **`autocorrect`** [global attribute](/en-US/docs/Web/HTML/Global_attributes) is an [enumerated](/en-US/docs/Glossary/Enumerated) attribute that controls whether editable text is automatically corrected for spelling and/or punctuation errors.

Autocorrection is relevant to editable text elements:

> 

Reviewer: same editing suggestions for the interface page above (for this section and the rest of the page)

Author: @estelle Thank you for the review- really appreciate your help. I have accepted most of it and argued a case or two. I got to the end and this is the only un-addressed issue. I am not sure whether I have addressed it in all the other comments or not. 

But I _think_ so. Can you re-review?

- {{htmlelement("input")}} elements, except for [`password`](/en-US/docs/Web/HTML/Element/input/password), [`email`](/en-US/docs/Web/HTML/Element/input/email), and [`url`](/en-US/docs/Web/HTML/Element/input/url), which do not support autocorrection.
- {{htmlelement("textarea")}} elements.
- Any element that has the [`contenteditable`](/en-US/docs/Web/HTML/Global_attributes/contenteditable) attribute set.

Editable elements have autocorrection enabled by default, except for within a {{htmlelement("form")}} element, where the default value may be inherited from the form.
Explicitly setting the attribute overrides the default.

## Value

Possible values are:

- `on` or `""` (the empty string)

  - : Enable automatic correction of spelling and punctuation errors.

- `off`

  - : Disable automatic correction of editable text.

> The {{htmlelement("input")}} element types that don't support auto-correction always have the `off` state: [`password`](/en-US/docs/Web/HTML/Element/input/password), [`email`](/en-US/docs/Web/HTML/Element/input/email) and [`url`](/en-US/docs/Web/HTML/Element/input/url).

Reviewer: safari is returning both true and false for password

Author: This is captured in BCD.

The original implementation was by Safari. This moved into the spec, and changed a little. Since we document the spec and other implementations are expected, this is captured in BCD.

> For all other editable elements, setting any value other than `off` is always treated as `on` (including the empty string), and the default value for elements that are not nested inside a `<form>` is `on`.

Reviewer: i've tried clarifying this sentence a few times, but don't like my version either. This paragraph is a bit difficut to parse

Author: Yes, it is horrible, and I spent a long time on it. 
Today this seem easier.

```suggestion
For all other editable elements, setting any other value than those listed above is always treated as `on`.
The default value for elements that are not nested inside a `<form>` is `on`.
```

Explicitly setting the attribute overrides the default.

## Value

Possible values are:

- `on` or `""` (the empty string)

  - : Enable automatic correction of spelling and punctuation errors.

- `off`

  - : Disable automatic correction of editable text.

The {{htmlelement("input")}} element types that don't support auto-correction always have the `off` state: [`password`](/en-US/docs/Web/HTML/Element/input/password), [`email`](/en-US/docs/Web/HTML/Element/input/email) and [`url`](/en-US/docs/Web/HTML/Element/input/url).

For all other editable elements, setting any other value than those listed above is always treated as `on`.
The default value for elements that are not nested inside a `<form>` is `on`.

When nested in a `<form>`, the following elements inherit their their default value of `autocorrect` from the form if it has been set: {{htmlelement("button")}}, {{htmlelement("fieldset")}}, {{htmlelement("input")}}, {{htmlelement("output")}}, {{htmlelement("select")}}, and {{htmlelement("textarea")}}.

## Examples

> 

Reviewer: ```suggestion
### Basic example

This example demonstrates basic `autocorrect` attribute usage.

#### HTML

We include two text `<input>` elements with different values for their `autocorrect` attributes:

```html
<label for="vegetable">A vegetable: </label>
<input id="vegetable" name="vegetable" type="text" autocorrect="on" />

<label for="fruit">A fruit: </label>
<input id="fruit" name="fruit" type="text" autocorrect="off" />
```end

#### Results

{{EmbedLiveSample("Basic example", "100%", "250")}}

Enter invalid text into the fruit and vegetable text entry boxes above.
If auto-correction is enabled on your browser, a typo in a vegetable name should be auto-corrected, but not in a fruit name.

```
